mtags/src/main/scala/scala/meta/internal/metals/TrigramSubstrings.scala

```diff
@@ -1,34 +1,40 @@
 package scala.meta.internal.metals
 
 import java.lang.StringBuilder
+import scala.collection.mutable.ArrayBuffer
 
-/**
- * Returns all possible substrings of length 3 for the given string.
- */
-class TrigramSubstrings(string: String) extends Traversable[String] {
-  override def foreach[U](f: String => U): Unit = {
-    val N = string.length
-    val arr = new Array[Char](3)
-    var i = 0
-    while (i < N) {
-      var j = i + 1
-      while (j < N) {
-        var k = j + 1
-        while (k < N) {
-          arr(0) = string.charAt(i)
-          arr(1) = string.charAt(j)
-          arr(2) = string.charAt(k)
-          f(new String(arr))
-          k += 1
+object TrigramSubstrings {
+
+  /**
+   * Returns all possible substrings of length 3 for the given string.
+   */
+  def apply(string: String): ArrayBuffer[String] = {
```

Comment:
Hmm, I'm wondering if we should do instead 

```scala
object TrigramSubstrings {
  def foreach(string: String, fn: String => Unit): Unit = ...
  def seq(string: String): Seq[String] = {
    val buf = ArrayBuffer.empty[String]
    foreach(string, trigram => buf += trigram)
    buf.result()
  }
}
```

It looks like we don't need the intermediary buffer in `Fuzzy`. This is a **very** hot path.